src/e2e-test/java/io/cdap/plugin/common/ValidationHelper.java

```diff
@@ -0,0 +1,278 @@
+
+    if (isMatched) {
+      System.out.println("The data is matched.");
+    } else {
+      System.out.println("The data is not matched.");
+    }
+
+    return isMatched;
+  }
+
+  public static boolean validateGCSDataToBQ(String bucketName, String table) throws IOException,
+    InterruptedException {
+    Map<String, JsonObject> expectedData = new HashMap<>();
+    Map<String, JsonObject> actualData = listBucketObjects(bucketName);
+
+    getBigQueryTableData(table, expectedData);
+
+    boolean isMatched = matchJsonMaps(expectedData, actualData);
+
+    if (isMatched) {
+      System.out.println("The data is matched.");
+    } else {
+      System.out.println("The data is not matched.");
+    }
+
+    return isMatched;
+  }
+
+  public static Map<String, JsonObject> listBucketObjects(String bucketName) {
+    Map<String, JsonObject> dataMap = new HashMap<>();
+    Storage storage = StorageOptions.newBuilder().setProjectId(PROJECT_ID).build().getService();
+    Page<Blob> blobs = storage.list(bucketName);
+
+    List<Blob> bucketObjects = StreamSupport.stream(blobs.iterateAll().spliterator(), true)
+      .filter(blob -> blob.getSize() != 0)
+      .collect(Collectors.toList());
+
+    Stream<String> objectNamesWithData = bucketObjects.stream().map(blob -> blob.getName());
+    List<String> bucketObjectNames = objectNamesWithData.collect(Collectors.toList());
+
+    if (!bucketObjectNames.isEmpty()) {
+      String objectName = bucketObjectNames.get(0);
+      if (objectName.contains("part-r")) {
+        Map<String, JsonObject> dataMap2 = fetchObjectData(PROJECT_ID, bucketName, objectName);
+        dataMap.putAll(dataMap2);
+      }
+    }
+
+    return dataMap;
+  }
+
+  private static Map<String, JsonObject> fetchObjectData(String projectId, String bucketName, String objectName) {
+    Map<String, JsonObject> dataMap = new HashMap<>();
+    Storage storage = StorageOptions.newBuilder().setProjectId(projectId).build().getService();
+    byte[] objectData = storage.readAllBytes(bucketName, objectName);
+    String objectDataAsString = new String(objectData, StandardCharsets.UTF_8);
+
+    // Splitting using the delimiter as a File can have more than one record.
+    String[] lines = objectDataAsString.split("\n");
+    for (String line : lines) {
+      JsonObject json = gson.fromJson(line, JsonObject.class);
+      String id = json.get("transaction_uid").getAsString();
+      dataMap.put(id, json);
+    }
+
+    return dataMap;
+  }
+
+  private static void getBigQueryTableData(String table, Map<String, JsonObject> bigQueryRows)
+    throws IOException, InterruptedException {
+
+    String projectId = PluginPropertyUtils.pluginProp("projectId");
+    String dataset = PluginPropertyUtils.pluginProp("dataset");
+    String selectQuery = "SELECT TO_JSON(t) FROM `" + projectId + "." + dataset + "." + table + "` AS t";
+    TableResult result = BigQueryClient.getQueryResult(selectQuery);
+    for (FieldValueList row : result.iterateAll()) {
```

Comment:
Ill suggest doing this in separate method, anyway what are we doing here using this logic ? If you simply want to add the json in a list there is a simple way for that.

Reply:
As we are retrieving all the corresponding rows from the big query table and iterating them based on the unique key and putting the data in a map.